app/scripts/inpage.js

```diff
@@ -22,6 +22,15 @@ var metamaskStream = new LocalMessageDuplexStream({
 // compose the inpage provider
 var inpageProvider = new MetamaskInpageProvider(metamaskStream)
 
+// Augment the provider with its enable method
+inpageProvider.enable = function () {
+  return new Promise((resolve) => {
+    resolve(inpageProvider.send({ method: 'eth_accounts' }).result)
```

Comment:
Does `inpageProvider.send({ method: 'eth_accounts' })` return a synchronous result, for grabbing `.result` off of?

Reply:
I think I would rather use the async method and return that method, these sync methods historically are unreliable.